marshal/writer.go

```diff
@@ -225,7 +225,9 @@ func (w *Writer) Uint8slVal(v reflect.Value) {
 	l := v.Len()
 	if l > 0 {
 		if v.Index(0).CanAddr() {
-			p := (*[gg]uint8)(unsafe.Pointer(v.Index(0).Addr().Pointer()))
+                        _p := uintptr( (unsafe.Pointer(v.Index(0).Addr().Pointer())) )
+                        sh := &reflect.SliceHeader{Data: _p, Len:  l, Cap:  l, }
```

Comment:
Unfortunatelly, reflect.SliceHeader is not suitable for practical use: conversion of pointer to uintptr and back is not safe with precise GC. Therefore custom slice header should be used:
```go
type sliceHeader struct { p unsafe.Pointer; l, c int }
```

And even call to `v.Index(0).Addr().Pointer()` is not safe :-(
But rather conversion to pointer should be done with type cast: `v.Index(0).Addr().Interface().(*uint8)`.

:-(